src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -223,10 +223,14 @@ protected virtual string BuildChallengeUrl(AuthenticationProperties properties,
             return QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, parameters);
         }
 
+        [Obsolete("Subclasses should override FormatScope(IEnumerable<string>) instead")]
         protected virtual string FormatScope()
+            => FormatScope(Options.Scope);
+
+        protected virtual string FormatScope(IEnumerable<string> scopes)
         {
             // OAuth2 3.3 space separated
-            return string.Join(" ", Options.Scope);
+            return string.Join(" ", scopes);
```

Comment:
nit: =>

Reply:
What do we do with comments for expression-bodied methods? Keep them in the line above the `=>`? I think it reads easier using “classic” method syntax here.

Reply:
Yeah either of these seems fine:

```C#
protected virtual string FormatScope(IEnumerable<string> scopes)
    // OAuth2 3.3 space separated
    => string.Join(" ", scopes);
```

```C#
protected virtual string FormatScope(IEnumerable<string> scopes)
    => string.Join(" ", scopes); // OAuth2 3.3 space separated
```

Reply:
✔️